graspologic/plot/plot_matrix.py

```diff
@@ -0,0 +1,969 @@
+                # map each node from the sorting class to their sorting class size
+                meta[f"{gc}_size_order"] = meta[gc].map(class_size)
+                total_sort_by.append(f"{gc}_size_order")
+            else:
+                class_value = meta.groupby(gc)[co].mean()
+                # map each node from the sorting class to certain sorting class attribute
+                meta[f"{gc}_{co}_order"] = meta[gc].map(class_value)
+                total_sort_by.append(f"{gc}_{co}_order")
+        total_sort_by.append(gc)
+    total_sort_by += item_order
+
+    # arbitrarily sort the data from 1 to length
+    meta["sort_idx"] = range(len(meta))
+    # if we actually need to sort, sort by class_order, group_class, item_order
+    if len(total_sort_by) > 0:
+        meta.sort_values(total_sort_by, inplace=True, kind="mergesort")
+
+    inds = np.copy(meta["sort_idx"].values)
+    meta["sort_idx"] = range(len(meta))
+    return inds, meta
+
+
+def matrixplot(
+    data,
+    ax=None,
+    col_meta=None,
+    row_meta=None,
+    plot_type="heatmap",
+    col_group_class=None,
+    row_group_class=None,
+    col_class_order="size",
+    row_class_order="size",
+    col_item_order=None,
+    row_item_order=None,
+    col_color_class=None,
+    row_color_class=None,
+    col_highlight=None,
+    row_highlight=None,
+    col_palette="tab10",
+    row_palette="tab10",
+    col_ticks=True,
+    row_ticks=True,
+    col_tick_pad=None,
+    row_tick_pad=None,
+    col_color_pad=None,
+    row_color_pad=None,
+    border=True,
+    center=0,
+    cmap="RdBu_r",
+    sizes=(5, 10),
+    square=True,
+    gridline_kws=None,
+    spinestyle_kws=None,
+    highlight_kws=None,
+    **kws,
+):
+    """
+    Sorts a matrix and plots it with ticks and colors on the borders
+
+    Parameters
+    ----------
+    data : np.ndarray, ndim=2
+        Matrix to plot
+    ax : matplotlib axes object, optional
+        Axes in which to draw the plot, by default None
+    plot_type : str, optional
+        One of "heatmap" or "scattermap", by default "heatmap"
+    {col, row}_meta : pd.DataFrame, pd.Series, list of pd.Series or np.array, optional
+        Metadata of the matrix such as class, cell type, etc., by default None
+    {col, row}_group_class : string or list of string, optional
+        Attribute in meta to group the graph in the plot, by default None
+    {col, row}_class_order : string or list of string, optional
+        Attribute of the sorting class to sort classes within the graph, by default "size"
+    {col, row}_item_order : string or list of string, optional
+        Attribute in meta by which to sort elements within a class, by default None
+    {col, row}_color_class : string or list of string, optional
+        Attribute in meta by which to draw colorbars, by default None
+    {col, row}_highlights : string or list of string, optional
+        Attribute in meta by which to draw highlighted separateors, by default None
```

Comment:
in my original implementation, these could also be passed as lists (like a list of labels) is that still possible?

Reply:
Do you mean a list of strings or something else?

Reply:
one other possible input is "array-like". more or less just means {list, np.ndarray, pd.Series}